src/awkward/_v2/index.py

```diff
@@ -27,6 +27,9 @@ def __init__(self, data, metadata=None, nplike=None):
         if len(self._data.shape) != 1:
             raise ak._v2._util.error(TypeError("Index data must be one-dimensional"))
 
+        if self._data.dtype.type == np.longlong:
+            self._data = self._data.astype(np.int64)
```

Comment:
Since longlong is just another word for 64-bit (or 32-bit?) integers, we don't need `astype`, which converts. It should be enough to use `view`, which is zero-copy.

In fact, we'd like to know if there are any systems for which longlong is not 64-bit, so we can put in an assertion with a question and hope someone answers it. Oh, [longlong is _at least_ 64-bits](https://stackoverflow.com/questions/5221615/c-long-long-always-64-bit), but my NumPy, at least, doesn't have an `np.int128` type.

Finally, `np.longlong` is a type, so we want to check for it with `issubclass`.

This should apply all of those:

```suggestion
        if issubclass(self._data.dtype.type, np.longlong):
            assert np.dtype(np.longlong).itemsize == 8, "longlong is always 64-bit, right?"
            self._data = self._data.view(np.int64)
```